tests/app/questionnaire/test_completeness.py

```diff
@@ -0,0 +1,483 @@
+
+    def test_get_first_incomplete_location_in_survey_completed(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+        completed_blocks = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+            Location('toast-group', 0, 'toast'),
+        ]
+        routing_path = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+            Location('toast-group', 0, 'toast'),
+        ]
+
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+        self.assertFalse(progress.get_first_incomplete_location_in_survey())
+
+    def get_first_incomplete_location_in_survey_completed_skipped_interstitials(self):
+        schema = load_schema_from_params('test', 'navigation')
+
+        # interstitial blocks have been removed
+        completed_blocks = [
+            Location('property-details', 0, 'insurance-type'),
+            Location('property-details', 0, 'insurance-address'),
+            Location('house-details', 0, 'house-type'),
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 0, 'repeating-block-2'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('payment-details', 0, 'credit-card'),
+            Location('payment-details', 0, 'expiry-date'),
+            Location('payment-details', 0, 'security-code'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+        ]
+
+        routing_path = [
+            Location('property-details', 0, 'insurance-type'),
+            Location('property-details', 0, 'insurance-address'),
+            Location('property-details', 0, 'property-interstitial'),
+            Location('house-details', 0, 'house-type'),
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 0, 'repeating-block-2'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('extra-cover', 0, 'extra-cover-interstitial'),
+            Location('payment-details', 0, 'credit-card'),
+            Location('payment-details', 0, 'expiry-date'),
+            Location('payment-details', 0, 'security-code'),
+            Location('payment-details', 0, 'security-code-interstitial'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+            Location('confirmation-group', 0, 'confirmation'),
+        ]
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+        self.assertFalse(progress.get_first_incomplete_location_in_survey())
+
+    def test_get_latest_location_no_completed_blocks(self):
```

Comment:
I wonder if we should be specifying what we expect on these tests?
`test_get_latest_location_no_completed_blocks_returns_first_location`

Reply:
I find that can result in really long method names. Prefer that the method name describes the scenario and the assertion should indicate the expected outcome.

Thoughts? @jonnyshaw89

Reply:
Yeh, i'm not die hard either way.
Our code is scattered with examples of both.